core/src/docs/upgrade.md

```diff
@@ -1,3 +1,12 @@
+# Upgrade to v0.33
+
+OpenDAL 0.33 redesigned the `Writer` API.
```

Comment:
Please introduce as much as possible about the changes in `writer.write` and the difference between it and `op.write`.

Reply:
DONE, please take a look again.

> the difference between it and `op.write`.

They are 100% differnt public APIs from day one of OpenDAL. There is no `difference` to discuss here.